3rdparty/glog-0.4.0.patch

```diff
@@ -0,0 +1,244 @@
++#endif
++
+ #ifdef GLOG_STL_LOGGING_FOR_UNORDERED
+ # include <unordered_map>
+ # include <unordered_set>
+diff --git a/src/glog/logging.h.in b/src/glog/logging.h.in
+index d6cc865..e90afab 100644
+--- a/src/glog/logging.h.in
++++ b/src/glog/logging.h.in
+@@ -1422,6 +1422,16 @@ class GOOGLE_GLOG_DLL_DECL LogSink {
+   // Sink's logging logic (message_len is such as to exclude '\n' at the end).
+   // This method can't use LOG() or CHECK() as logging system mutex(s) are held
+   // during this call.
++  virtual void send(LogSeverity severity, const char* full_filename,
++                    const char* base_filename, int line,
++                    const struct ::tm* tm_time,
++                    const char* message, size_t message_len, int usec) {
++    send(severity, full_filename, base_filename, line,
++         tm_time, message, message_len);
++  }
++  // This send() signature is obsolete.
++  // New implementations should define this in terms of
++  // the above send() method.
+   virtual void send(LogSeverity severity, const char* full_filename,
+                     const char* base_filename, int line,
+                     const struct ::tm* tm_time,
+@@ -1446,7 +1456,15 @@ class GOOGLE_GLOG_DLL_DECL LogSink {
+   // Can be useful to implement send().
+   static std::string ToString(LogSeverity severity, const char* file, int line,
+                               const struct ::tm* tm_time,
+-                              const char* message, size_t message_len);
++                              const char* message, size_t message_len,
++                              int usecs);
++
++  // Obsolete
++  static std::string ToString(LogSeverity severity, const char* file, int line,
++                              const struct ::tm* tm_time,
++                              const char* message, size_t message_len) {
++    return ToString(severity, file, line, tm_time, message, message_len ,0);
++  }
+ };
+ 
+ // Add or remove a LogSink as a consumer of logging data.  Thread-safe.
+diff --git a/src/logging.cc b/src/logging.cc
+index 0b5e6ee..9f72fe5 100644
+--- a/src/logging.cc
++++ b/src/logging.cc
+@@ -351,6 +351,7 @@ struct LogMessage::LogMessageData  {
+   };
+   time_t timestamp_;            // Time of creation of LogMessage
+   struct ::tm tm_time_;         // Time of creation of LogMessage
++  int usecs_;                   // Time of creation of LogMessage - microseconds part
+   size_t num_prefix_chars_;     // # of chars of prefix in this message
+   size_t num_chars_to_log_;     // # of chars of msg to send to log
+   size_t num_chars_to_syslog_;  // # of chars of msg to send to syslog
+@@ -516,7 +517,8 @@ class LogDestination {
+                          int line,
+                          const struct ::tm* tm_time,
+                          const char* message,
+-                         size_t message_len);
++                         size_t message_len,
++                         int usecs);
```

Comment:
I think we need at least `long` for millionths of a second. (although Mesos itself doesn't really support exotic architectures, there's at least one guy experimenting with it on raspberry pi and other embedded things, so I think it's nicer for us to avoid porting issues when we spot them)

Reply:
Since C++11 the right type would be int_least32_t or probably int_fast32_t.
I'm not really sure if glog still has to support pre-C++11 standard...

Reply:
Glog is already defining google::int32, switched to use that.